Refactor email code and templates

karenc opened the pull request:
- Remove wrong comments and NOQA imports from email extensions and model


- Refactor email resolve recipient and test code

  Resolve recipients when the Email object is being created in `__init__`.
  Move email set up code to an autouse fixture to make sure emails are not
  being sent.

- Refactor email templates into a base template

  - Move most of the html in the email templates into
    `app/templates/email/en_us/base.jinja2`
  - Add tests for all the email templates
  - Add default in `SiteSetting.HOUSTON_SETTINGS`

hwindsor commented on app/extensions/email/__init__.py:
```diff
@@ -184,6 +167,32 @@ def template(self, template, **kwargs):
+            'site_name', SiteSetting.get_value('site.name', default='Codex')
+        )
+        kwargs.setdefault('site_url', '/')
+        kwargs.setdefault('site_domain', 'flukebook.org')
```
That looks wrong.

karenc replied:
Let me see where we have the domain setting...

karenc replied:
```diff
@@ -186,7 +196,10 @@ class Email(Message):
             'site_name', SiteSetting.get_value('site.name', default='Codex')
         )
         kwargs.setdefault('site_url', '/')
-        kwargs.setdefault('site_domain', 'flukebook.org')
+        kwargs.setdefault(
+            'site_domain',
+            urlparse(current_app.config.get('BASE_URL')).netloc.split(':')[0],
+        )
         kwargs.setdefault('twitter_url', SiteSetting.get_value('twitter_url'))
         kwargs.setdefault('facebook_url', SiteSetting.get_value('facebook_url'))
         kwargs.setdefault(
```

naknomum replied:
this seems to be a fussy thing to figure out.  i had tried to basically come up with code to get "the site url" here:
https://github.com/WildMeOrg/houston/blob/main/app/utils.py#L42-L53

i am not sure the best way to make this work in both a production and dev environment.    i am curious to see if your code above does the expected thing on, say, the QA server (which managed to not work at all with my code).

karenc replied:
Thanks, I'm using `site_url_prefix()` instead of `BASE_URL`.  I changed `site_url_prefix()` to return `localhost:84` instead so it's the same as the development set up

hwindsor commented on app/templates/email/en_us/notifications/collaboration_approved.jinja2:
```diff
@@ -1,525 +1,7 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
-<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" style="background:#f3f3f3!important"
+{% extends 'email/en_us/base.jinja2' %}
+{% block title %}Collaboration Approved, Flukebook{% endblock %}
```
This is generally MUCH more readable and is a big improvement, Thanks for this. 
But there  are a few flukebook specific references, this is one of them and needs to go : )

hwindsor replied:
Not sure, I was goinng by what was in line 4

karenc replied:
By the way I'm not sure where `<title>Collaboration Approved, Flukebook</title>` actually is shown in the email...

![Screenshot 2022-02-24 at 16-32-59 You have a collaboration request from Houston on context_not_set - karen karen-chan com -](https://user-images.githubusercontent.com/153353/155566826-34ad9e38-cd93-45d4-83ba-45c0c33ff55f.png)

hwindsor replied:
Interesting, also not clear is where that flukebook image across the top comes from :)

karenc replied:
Yeah the title is in the html but is not displayed by the gmail web client.  The flukebook image is from the `header_image_url`.

hwindsor commented on app/modules/site_settings/models.py:
```diff
@@ -47,6 +47,49 @@ class SiteSetting(db.Model, Timestamp):
         'email_service_password': {'type': str, 'public': False},
         'email_default_sender_email': {'type': str, 'public': False},
         'email_default_sender_name': {'type': str, 'public': False},
+        'email_header_image_url': {
```
Defaults need to be non flukebook specific, but otherwise, yes very much of an improvement.

karenc replied:
I thought we just need an example so people know what is expected.  The default shouldn't be used obviously :sweat_smile:

hwindsor replied:
Understood. The danger with that is that if you leave defaults that are valid for one project, they might get inadertently used by another and that would cause problems to debug.

karenc replied:
I think we need to have a send test email API and / or an API to show the basic email template so people knows what it'll look like

hwindsor replied:
Oh I agree with that entirely. All I'm saying is that noneof these defaults should be specific to any platform and currently they are.

hwindsor replied:
Oh I agree with that entirely. All I'm saying is that noneof these defaults should be specific to any platform and currently they are.

colinwkingen replied:
We have a generic Wildbook logo, but not one for Codex yet. Perhaps the Wild Me logo could be used, that's an existing asset.

hwindsor commented on app/modules/site_settings/models.py:
```diff
@@ -47,6 +47,49 @@ class SiteSetting(db.Model, Timestamp):
         'email_service_password': {'type': str, 'public': False},
         'email_default_sender_email': {'type': str, 'public': False},
         'email_default_sender_name': {'type': str, 'public': False},
+        'email_header_image_url': {
```
Defaults need to be non flukebook specific, but otherwise, yes very much of an improvement.

naknomum replied:
many of these things are set up via the configuration that comes from edm (which pre-dates SiteSettings) and their values (and defaults, even) should come from there.    i am going to take this to slack....

karenc commented:
@hwindsor @naknomum Changes made:

 - Added `MAIL_BASE_URL` environment variable (it's already in the config) for development in order to get the emails to have localhost:84 instead of houston:5000, this isn't a problem for production
 - Moved the template arguments into `Email.__init__`
 - Removed all "whale" or "flukebook" from code and templates
 - Changed the year in legal statement to 2022
 - Removed `twitter_url` and `facebook_url` from `SiteSetting` and use the settings from edm instead
 - Added instagram url to the email template
 - Only display facebook, twitter, instagram buttons if urls are set
 - Changed `app.utils.site_url_prefix` to return `localhost:84` instead of `localhost`

<details>

<summary>Click here to see the diff</summary>

```diff
diff --git a/.env b/.env
index 033d6c4e..ebe2588d 100644
--- a/.env
+++ b/.env
@@ -42,6 +42,7 @@ HOUSTON_DB_NAME=houston
 HOUSTON_DB_USER=houston
 HOUSTON_DB_PASSWORD=development
 HOUSTON_URL=http://houston:5000/
+MAIL_BASE_URL=http://localhost:84/
 WILDBOOK_DB_HOST=db
 
 #: i.e. postgresql://${HOUSTON_DB_USER}:${HOUSTON_DB_PASSWORD}@db/${HOUSTON_DB_NAME}
diff --git a/app/extensions/email/__init__.py b/app/extensions/email/__init__.py
index 0c4d788d..7b5864b8 100644
--- a/app/extensions/email/__init__.py
+++ b/app/extensions/email/__init__.py
@@ -1,6 +1,10 @@
 # -*- coding: utf-8 -*-
 # pylint: disable=no-self-use
+import datetime
+from io import StringIO
 import logging
+import re
+from urllib.parse import urlparse
 
 from flask import current_app, render_template
 from flask_mail import Mail, Message, email_dispatched
@@ -11,12 +15,6 @@ import htmlmin
 import app.version
 from app.utils import to_ascii
 
-from io import StringIO
-import re
-
-import datetime
-
-
 from flask_restx_patched import is_extension_enabled
 
 if not is_extension_enabled('mail'):
@@ -133,6 +131,18 @@ class Email(Message):
             'site_url_prefix': site_url_prefix(),
             'year': now.year,
             'transaction_id': self._transaction_id,
+            'header_image_url': SiteSetting.get_value('email_header_image_url'),
+            'h1': SiteSetting.get_value('email_title_greeting'),
+            'secondary_title': SiteSetting.get_value('email_secondary_title'),
+            'secondary_text': SiteSetting.get_value('email_secondary_text'),
+            'legal_statement': SiteSetting.get_value('email_legal_statement'),
+            'unsubscribe_link': '/unsubscribe',
+            'site_url': '/',
+            'site_domain': urlparse(site_url_prefix()).netloc,
+            'instagram_url': SiteSetting.get_value('site.links.instagramLink'),
+            'twitter_url': SiteSetting.get_value('site.links.twitterLink'),
+            'facebook_url': SiteSetting.get_value('site.links.facebookLink'),
+            'adoption_button_text': SiteSetting.get_value('email_adoption_button_text'),
         }
         self.status = None
         self.mail = mail
@@ -167,32 +177,6 @@ class Email(Message):
             pmail = Premailer(**pmail_kwargs)  # REF: https://pypi.org/project/premailer/
         log.info('Using premailer = %r' % (pmail))
 
-        from app.modules.site_settings.models import SiteSetting
-
-        # Default values for templates
-        kwargs.setdefault(
-            'header_image_url', SiteSetting.get_value('email_header_image_url')
-        )
-        kwargs.setdefault('h1', SiteSetting.get_value('email_title_greeting'))
-        kwargs.setdefault(
-            'secondary_title', SiteSetting.get_value('email_secondary_title')
-        )
-        kwargs.setdefault('secondary_text', SiteSetting.get_value('email_secondary_text'))
-        kwargs.setdefault(
-            'legal_statement', SiteSetting.get_value('email_legal_statement')
-        )
-        kwargs.setdefault('unsubscribe_link', '/unsubscribe')
-        kwargs.setdefault(
-            'site_name', SiteSetting.get_value('site.name', default='Codex')
-        )
-        kwargs.setdefault('site_url', '/')
-        kwargs.setdefault('site_domain', 'flukebook.org')
-        kwargs.setdefault('twitter_url', SiteSetting.get_value('twitter_url'))
-        kwargs.setdefault('facebook_url', SiteSetting.get_value('facebook_url'))
-        kwargs.setdefault(
-            'adoption_button_text', SiteSetting.get_value('email_adoption_button_text')
-        )
-
         self.template_name = template
         self.template_kwargs.update(kwargs)
         self._template_found = False
diff --git a/app/modules/site_settings/models.py b/app/modules/site_settings/models.py
index 8f67c817..8050810b 100644
--- a/app/modules/site_settings/models.py
+++ b/app/modules/site_settings/models.py
@@ -50,7 +50,6 @@ class SiteSetting(db.Model, Timestamp):
         'email_header_image_url': {
             'type': str,
             'public': True,
-            'default': 'http://www.flukebook.org/cust/mantamatcher/img/flukebook_header.png',
         },
         'email_title_greeting': {
             'type': str,
@@ -60,35 +59,25 @@ class SiteSetting(db.Model, Timestamp):
         'email_secondary_title': {
             'type': str,
             'public': True,
-            'default': 'Whale whales for everyone!',
+            'default': 'Codex for everyone!',
         },
         'email_secondary_text': {
             'type': list,
             'public': True,
             'default': [
-                'Can you help researchers study and protect whales?',
+                'Can you help researchers study and protect animals?',
                 'Wild Me is a 501(c)(3) non-profit that develops the cutting edge technology for a global community of wildlife researchers. Are you interested in further supporting our mission?',
             ],
         },
         'email_adoption_button_text': {
             'type': str,
             'public': True,
-            'default': 'Adopt a whale',
+            'default': 'Adopt an animal',
         },
         'email_legal_statement': {
             'type': str,
             'public': True,
-            'default': 'Wildbook is copyrighted 2016 by Wild Me.  Individual contributors to Wildbook retain copyrights for submitted images. No reproduction or usage of material in this library is permitted without the express permission of the copyright holders.',
-        },
-        'twitter_url': {
-            'type': str,
-            'public': True,
-            'default': 'https://twitter.com/IBEIS_org',
-        },
-        'facebook_url': {
-            'type': str,
-            'public': True,
-            'default': 'https://www.facebook.com/wildmeorg/',
+            'default': 'Wildbook is copyrighted 2022 by Wild Me.  Individual contributors to Wildbook retain copyrights for submitted images. No reproduction or usage of material in this library is permitted without the express permission of the copyright holders.',
         },
         'social_group_roles': {'type': list, 'public': True},
         'relationship_type_roles': {
diff --git a/app/templates/email/en_us/base.jinja2 b/app/templates/email/en_us/base.jinja2
index 36a6eba1..3dff948b 100644
--- a/app/templates/email/en_us/base.jinja2
+++ b/app/templates/email/en_us/base.jinja2
@@ -37,11 +37,11 @@
                                       <table style="border-collapse:collapse;border-spacing:0;padding:0;text-align:left;vertical-align:top;width:100%">
                                         <tbody>
                                           <tr style="padding:0;text-align:left;vertical-align:top">
-                                            <th style="Margin:0;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;padding:0;text-align:left"><img
+                                            <th style="Margin:0;color:#ffffff;font-family:Helvetica,Arial,sans-serif;font-size:34px;font-weight:400;line-height:1.3;margin:0;padding:0;text-align:left">{% if header_image_url %}<img
 
-                                                src="{{ header_image }}"
+                                                src="{{ header_image_url }}"
 
-                                                style="-ms-interpolation-mode:bicubic;clear:both;display:block;max-width:100%;outline:0;text-decoration:none;width:auto" /></th>
+                                                style="-ms-interpolation-mode:bicubic;clear:both;display:block;max-width:100%;outline:0;text-decoration:none;width:auto" />{% else %}{{ site_name }}{% endif %}</th>
                                           </tr>
                                         </tbody>
                                       </table>
@@ -278,6 +278,7 @@
                                                             <th style="Margin:0;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;padding:0;text-align:left">
                                                               <h5 style="Margin:0;Margin-bottom:10px;color:inherit;font-family:Helvetica,Arial,sans-serif;font-size:20px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left;word-wrap:normal">Connect
                                                                 With Us:</h5>
+                                                              {% if facebook_url %}
                                                               <table class="button facebook expand"
 
                                                                 style="Margin:0 0 16px 0;border-collapse:collapse;border-spacing:0;margin:0 0 16px 0;padding:0;text-align:left;vertical-align:top;width:100%!important">
@@ -311,6 +312,8 @@
                                                                   </tr>
                                                                 </tbody>
                                                               </table>
+                                                              {% endif %}
+                                                              {% if twitter_url %}
                                                               <table class="button twitter expand"
 
                                                                 style="Margin:0 0 16px 0;border-collapse:collapse;border-spacing:0;margin:0 0 16px 0;padding:0;text-align:left;vertical-align:top;width:100%!important">
@@ -344,6 +347,42 @@
                                                                   </tr>
                                                                 </tbody>
                                                               </table>
+                                                              {% endif %}
+                                                              {% if instagram_url %}
+                                                              <table class="button instagram expand"
+
+                                                                style="Margin:0 0 16px 0;border-collapse:collapse;border-spacing:0;margin:0 0 16px 0;padding:0;text-align:left;vertical-align:top;width:100%!important">
+                                                                <tbody>
+                                                                  <tr style="padding:0;text-align:left;vertical-align:top">
+                                                                  <td style="-moz-hyphens:auto;-webkit-hyphens:auto;Margin:0;border-collapse:collapse!important;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;hyphens:auto;line-height:1.3;margin:0;padding:0;text-align:left;vertical-align:top;word-wrap:break-word">
+                                                                  <table style="border-collapse:collapse;border-spacing:0;padding:0;text-align:left;vertical-align:top;width:100%">
+                                                                  <tbody>
+                                                                  <tr style="padding:0;text-align:left;vertical-align:top">
+                                                                  <td style="-moz-hyphens:auto;-webkit-hyphens:auto;Margin:0;background:#F56040!important;border:2px solid #F56040;border-collapse:collapse!important;color:#fefefe;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;hyphens:auto;line-height:1.3;margin:0;padding:0;text-align:left;vertical-align:top;word-wrap:break-word">
+                                                                  <center data-parsed=""
+
+                                                                  style="min-width:0;width:100%"><a
+
+                                                                  href="{{ instagram_url }}"
+
+                                                                  align="center"
+
+                                                                  class="float-center"
+
+                                                                  style="Margin:0;border:0 solid #2199e8;border-radius:3px;color:#fefefe;display:inline-block;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:700;line-height:1.3;margin:0;padding:8px 16px 8px 16px;padding-left:0;padding-right:0;text-align:center;text-decoration:none;width:100%">Instagram</a></center>
+                                                                  </td>
+                                                                  </tr>
+                                                                  </tbody>
+                                                                  </table>
+                                                                  </td>
+                                                                  <td class="expander"
+
+                                                                  style="-moz-hyphens:auto;-webkit-hyphens:auto;Margin:0;border-collapse:collapse!important;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;hyphens:auto;line-height:1.3;margin:0;padding:0!important;text-align:left;vertical-align:top;visibility:hidden;width:0;word-wrap:break-word"><br />
+                                                                  </td>
+                                                                  </tr>
+                                                                </tbody>
+                                                              </table>
+                                                              {% endif %}
                                                               <table class="button google expand"
 
                                                                 style="Margin:0 0 16px 0;border-collapse:collapse;border-spacing:0;margin:0 0 16px 0;padding:0;text-align:left;vertical-align:top;width:100%!important">
diff --git a/app/templates/email/en_us/notifications/collaboration_approved.jinja2 b/app/templates/email/en_us/notifications/collaboration_approved.jinja2
index 12544a35..e826a95e 100644
--- a/app/templates/email/en_us/notifications/collaboration_approved.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_approved.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Approved, Flukebook{% endblock %}
+{% block title %}Collaboration Approved, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">Your collaboration request on {{ context_name }} was approved by {{ sender_name }}.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_edit_approved.jinja2 b/app/templates/email/en_us/notifications/collaboration_edit_approved.jinja2
index ca384930..df93bfff 100644
--- a/app/templates/email/en_us/notifications/collaboration_edit_approved.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_edit_approved.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Approved, Flukebook{% endblock %}
+{% block title %}Collaboration Approved, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">Your collaboration edit request on {{ context_name }} was approved by {{ sender_name }}.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_edit_request.jinja2 b/app/templates/email/en_us/notifications/collaboration_edit_request.jinja2
index 4d3b505c..0d4c0da3 100644
--- a/app/templates/email/en_us/notifications/collaboration_edit_request.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_edit_request.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Request, Flukebook{% endblock %}
+{% block title %}Collaboration Request, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">You have a request to edit on your collaboration on {{ context_name }} with {{ sender_name }}.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_edit_revoke.jinja2 b/app/templates/email/en_us/notifications/collaboration_edit_revoke.jinja2
index 448c8df7..671a8032 100644
--- a/app/templates/email/en_us/notifications/collaboration_edit_revoke.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_edit_revoke.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Revoked, Flukebook{% endblock %}
+{% block title %}Collaboration Revoked, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">Your collaboration on {{ context_name }} with {{ sender_name }} has had edit permission revoked.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_manager_create.jinja2 b/app/templates/email/en_us/notifications/collaboration_manager_create.jinja2
index 9bb9e234..3c7a076d 100644
--- a/app/templates/email/en_us/notifications/collaboration_manager_create.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_manager_create.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Created, Flukebook{% endblock %}
+{% block title %}Collaboration Created, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">A collaboration has been created by {{ sender_name }} on {{ context_name }} between {{user1_name}} and {{user2_name}}.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_manager_revoke.jinja2 b/app/templates/email/en_us/notifications/collaboration_manager_revoke.jinja2
index e120fec7..4a901648 100644
--- a/app/templates/email/en_us/notifications/collaboration_manager_revoke.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_manager_revoke.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Revoked, Flukebook{% endblock %}
+{% block title %}Collaboration Revoked, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">The collaboration on {{ context_name }} between {{user1_name}} and {{user2_name}} has been revoked by {{ sender_name }}.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_request.jinja2 b/app/templates/email/en_us/notifications/collaboration_request.jinja2
index 635772ea..47ae0fb4 100644
--- a/app/templates/email/en_us/notifications/collaboration_request.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_request.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Request, Flukebook{% endblock %}
+{% block title %}Collaboration Request, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">You have a request to collaborate on {{ context_name }} from {{ sender_name }}.</p>
 
diff --git a/app/templates/email/en_us/notifications/collaboration_revoke.jinja2 b/app/templates/email/en_us/notifications/collaboration_revoke.jinja2
index 8b170e2e..938c9737 100644
--- a/app/templates/email/en_us/notifications/collaboration_revoke.jinja2
+++ b/app/templates/email/en_us/notifications/collaboration_revoke.jinja2
@@ -1,5 +1,5 @@
 {% extends 'email/en_us/base.jinja2' %}
-{% block title %}Collaboration Revoked, Flukebook{% endblock %}
+{% block title %}Collaboration Revoked, {{ site_name }}{% endblock %}
 {% block main %}
 <p style="Margin:0;Margin-bottom:10px;color:#0a0a0a;font-family:Helvetica,Arial,sans-serif;font-size:16px;font-weight:400;line-height:1.3;margin:0;margin-bottom:10px;padding:0;text-align:left">Your collaboration on {{ context_name }} with {{ sender_name }} has been revoked.</p>
 
diff --git a/app/utils.py b/app/utils.py
index b2e2635e..9e834cad 100644
--- a/app/utils.py
+++ b/app/utils.py
@@ -49,7 +49,7 @@ def site_url_prefix():
     host = current_app.config.get('SERVER_NAME', 'codex.example.com')
     if not scheme or not host:
         scheme = 'http'
-        host = 'localhost'
+        host = 'localhost:84'  # development
     return f'{scheme}://{host}'.lower()
 
 
diff --git a/config/codex.py b/config/codex.py
index 408603a9..3f0ee653 100644
--- a/config/codex.py
+++ b/config/codex.py
@@ -85,7 +85,7 @@ class ProductionConfig(BaseCodexConfig):
 
     BASE_URL = os.environ.get('HOUSTON_URL')
 
-    MAIL_BASE_URL = BASE_URL
+    MAIL_BASE_URL = os.environ.get('MAIL_BASE_URL', BASE_URL)
     MAIL_OVERRIDE_RECIPIENTS = None
     MAIL_ERROR_RECIPIENTS = [
         'mail-errors@wildme.org',
@@ -99,7 +99,7 @@ class DevelopmentConfig(BaseCodexConfig):
 
     BASE_URL = os.environ.get('HOUSTON_URL', 'https://wildme.ngrok.io/')
 
-    MAIL_BASE_URL = BASE_URL
+    MAIL_BASE_URL = os.environ.get('MAIL_BASE_URL', BASE_URL)
     MAIL_OVERRIDE_RECIPIENTS = [
         'testing@wildme.org',
     ]
diff --git a/docker-compose.codex.yml b/docker-compose.codex.yml
index 277a5c58..99eeba9d 100644
--- a/docker-compose.codex.yml
+++ b/docker-compose.codex.yml
@@ -158,6 +158,7 @@ services:
       ACM_AUTHENTICATIONS_URI__DEFAULT: "${ACM_AUTHENTICATIONS_URI__DEFAULT}"
       ELASTICSEARCH_HOSTS: "${ELASTICSEARCH_HOSTS}"
       HOUSTON_URL: "${HOUSTON_URL}"
+      MAIL_BASE_URL: "${MAIL_BASE_URL}"
       REDIS_HOST: redis
       REDIS_PASSWORD: "seekret_development_password"
       GITLAB_PROTO: "${GITLAB_PROTO}"
```

colinwkingen reviewed (approved):
LGTM. Thanks for making all those tweaks. Having 'Flukebook' drift back into the main repo was something that plagued us in past Wildbook.
